ci: Tagging AMI image to all regions fails

**Logs**

```
Waiting for import-snapshot task to finalize. 80s
Waiting for import-snapshot task to finalize. 90s
Tagging Snapshot
Register AMI from snapshot
Tagging AMI
Making AMI public
AMI Created: ami-04a3865a917c8c92b
Copying AMI ami-04a3865a917c8c92b to all regions
Tagging Copied AMI ami-09dbfb65a74fee1a5
Making AMI ami-09dbfb65a74fee1a5 public

An error occurred (InvalidAMIID.NotFound) when calling the ModifyImageAttribute operation: The image id '[ami-09dbfb65a74fee1a5]' does not exist
make: *** [make/Makefile.images:72: aws_vanilla_ami] Error 254
```


**Additional context**
https://github.com/rancher-sandbox/cOS-toolkit/runs/3153878978?check_suite_focus=true

https://github.com/rancher-sandbox/cOS-toolkit/blob/master/images/aws_upload.sh#L69

--- a/.github/build.yaml.gomplate
+++ b/.github/build.yaml.gomplate
@@ -586,6 +586,7 @@
         run: |
           export github_sha="${GITHUB_SHA}"
           export COPY_AMI_ALL_REGIONS="true"
+          export AWS_DISK_NAME="cOS-Vanilla-${{ env.COS_VERSION }}
           make aws_vanilla_ami
 {{{ end }}}
 

--- a/.github/workflows/build-master.yaml
+++ b/.github/workflows/build-master.yaml
@@ -1274,5 +1274,6 @@ jobs:
         run: |
           export github_sha="${GITHUB_SHA}"
           export COPY_AMI_ALL_REGIONS="true"
+          export AWS_DISK_NAME="cOS-Vanilla-${{ env.COS_VERSION }}
           make aws_vanilla_ami
 

--- a/images/aws_upload.sh
+++ b/images/aws_upload.sh
@@ -3,9 +3,9 @@
 set -e
 
 disk="$1"
-s3_bucket="cos-images"
-disk_name="cOS-Vanilla"
-disk_desc="cOS Vanilla Image"
+s3_bucket="${BUCKET:-cos-images}"
+disk_name="${AWS_DISK_NAME:-cOS-Vanilla}"
+disk_desc="${AWS_DISK_DESC:-cOS Vanilla Image}"
 disk_build_date=$(date +"%m%d%Y_%H%M%S")
 : "${github_sha:=none}"
 : "${COPY_AMI_ALL_REGIONS:=false}"
@@ -42,7 +42,7 @@ snap_id=$(aws ec2 describe-import-snapshot-tasks \
 
 echo "Tagging Snapshot"
 aws ec2 create-tags --resources "${snap_id}" \
-    --tags Key=Name,Value=${disk_name} Key=Project,Value=cOS Key=Git_SHA,Value=$git_sha Key=Flavor,Value=cos-vanilla
+    --tags Key=Name,Value=${disk_name} Key=Project,Value=cOS Key=Git_SHA,Value=$github_sha Key=Flavor,Value=cos-vanilla
 
 echo "Register AMI from snapshot"
 ami_id=$(aws ec2 register-image \
@@ -58,7 +58,7 @@ ami_id=$(aws ec2 register-image \
 
 echo "Tagging AMI"
 aws ec2 create-tags --resources "${ami_id}" \
-   --tags Key=Name,Value=${disk_name} Key=Project,Value=cOS Key=Git_SHA,Value=$git_sha Key=Flavor,Value=cos-vanilla
+   --tags Key=Name,Value=${disk_name} Key=Project,Value=cOS Key=Git_SHA,Value=$github_sha Key=Flavor,Value=cos-vanilla
 
 echo "Making AMI public"
 aws ec2 modify-image-attribute --image-id "${ami_id}" --launch-permission "Add=[{Group=all}]"
@@ -92,7 +92,7 @@ if [[ "${COPY_AMI_ALL_REGIONS}" == "true" ]]
       --tags Key=Name,Value=${disk_name} Key=Project,Value=cOS Key=GITHUB_SHA,Value=$github_sha Key=Flavor,Value=cos-vanilla
 
     echo "Making AMI ${ami_copy_id} public"
-    aws ec2 modify-image-attribute --image-id "${ami_copy_id}" --launch-permission "Add=[{Group=all}]"
+    aws ec2 modify-image-attribute --image-id "${ami_copy_id}" --region "${reg}" --launch-permission "Add=[{Group=all}]"
 
     echo "AMI Copied: ${ami_copy_id}"
   done

--- a/make/Makefile.images
+++ b/make/Makefile.images
@@ -4,6 +4,8 @@
 #
 
 RAW?=$(shell ls $(ROOT_DIR)/*.raw 2> /dev/null)
+AWS_DISK_NAME?=cOS-Vanilla
+AWS_DISK_DESC?=cOS Vanilla Image
 
 QEMUIMG?=$(shell which qemu-img 2> /dev/null)
 

